### PR TITLE
TEST: do local check in mpi tests

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Compute correct buf locally instead of rely on MPI.

### CODE REVIEW

**Comment by Sergei-Lebedev** [comment]
> @Sergei-Lebedev You might need to update the copyright on some files.

fixed

File: test/mpi/test_allgather.cc [hunk 1]
```diff
@@ -70,18 +71,16 @@ ucc_status_t TestAllgather::set_input(int iter_persistent)
 
 ucc_status_t TestAllgather::check()
 {
-    int size, completed;
-    MPI_Comm_size(team.comm, &size);
-    size_t       single_rank_count = args.dst.info.count / size;
-    MPI_Datatype mpi_dt            = ucc_dt_to_mpi(dt);
-    MPI_Request  req;
+    size_t dt_size, single_rank_count;
+    int size;
```
Comment (shimmybalsam) [inline test/mpi/test_allgather.cc:75]: alignment

File: test/mpi/test_allgather.cc [hunk 2]
```diff
@@ -70,18 +71,16 @@ ucc_status_t TestAllgather::set_input(int iter_persistent)
...
-    MPI_Iallgather(MPI_IN_PLACE, 0, MPI_DATATYPE_NULL, check_buf,
-                   single_rank_count, mpi_dt, team.comm, &req);
-    do {
-        MPI_Test(&req, &completed, MPI_STATUS_IGNORE);
-        ucc_context_progress(team.ctx);
-    } while(!completed);
+    MPI_Comm_size(team.comm, &size);
+    single_rank_count = args.dst.info.count / size;
+    dt_size = ucc_dt_size(dt);
+    for (int i = 0; i < size; i++) {
```
Comment (shimmybalsam) [inline test/mpi/test_allgather.cc:80]: declare int i at start of function

File: test/mpi/test_alltoall.cc [hunk 3]
```diff
@@ -99,19 +100,18 @@ ucc_status_t TestAlltoall::set_input(int iter_persistent)
...
     MPI_Comm_size(team.comm, &size);
     single_rank_count = args.src.info.count / size;
 
-    MPI_Ialltoall(MPI_IN_PLACE, 0, MPI_DATATYPE_NULL, check_buf,
-                  single_rank_count, ucc_dt_to_mpi(dt), team.comm, &req);
-    do {
-        MPI_Test(&req, &completed, MPI_STATUS_IGNORE);
-        ucc_context_progress(team.ctx);
-    } while(!completed);
+    for (int i = 0; i < size; i++) {
```
Comment (shimmybalsam) [inline test/mpi/test_alltoall.cc:110]: declare int i at start of function

**Review by shimmybalsam — APPROVED** [review]

**Review by samnordmann — APPROVED** [review]

**Review by manjugv — APPROVED** [review]

**Pull request merged**